modules/database/classes/Kohana/Database/MySQLi.php

```diff
@@ -173,6 +173,9 @@ public function query($type, $sql, $as_object = FALSE, array $params = NULL)
 				':query' => $sql
 			], $this->_connection->errno);
 		}
+		
+		do {}
+        	while (mysqli_more_results($this->_connection) AND mysqli_next_result($this->_connection));
```

Comment:
Using procedural style, should be like $this->_connection->more_results and $this->_connection->next_result

Reply:
Please prepare DB/code where this is useful and an error is thrown/returned.